frontend/src/components/order/conversationThread.tsx

```diff
@@ -0,0 +1,117 @@
+import { createMessage, MessagePublic } from "@/client";
+import { RoleTag } from "@/components/common/roleTag.tsx";
+import { getAuthorizationHeader } from "@/utils.ts";
+import { Box, Button, Checkbox, Flex, Link, Textarea } from "@chakra-ui/react";
+import { Link as RouterLink, useRouter } from "@tanstack/react-router";
+import { useEffect, useRef, useState } from "react";
+import { useAuth } from "react-oidc-context";
+
+interface Props {
+  messages: MessagePublic[];
+  orderId: number;
+}
+
+export function ConversationThread({ messages, orderId }: Props) {
+  const auth = useAuth();
+  const router = useRouter();
+
+  const [messageValue, setMessageValue] = useState<string>("");
+
+  const messageBoxRef = useRef<HTMLDivElement>(null);
+
+  const scrollToBottom = () => {
+    if (messageBoxRef.current) {
+      messageBoxRef.current.scrollTop = messageBoxRef.current.scrollHeight;
+    }
+  };
+
+  useEffect(() => {
+    scrollToBottom();
+  }, [messages]);
+
+  const handleSendMessage = async () => {
+    try {
+      await createMessage({
```

Comment:
`router.invalidate()` forces a full page re-render.
Use local state to append the new message.
E.g.
```
try {
    const newMessage = await createMessage({
      headers: { ...getAuthorizationHeader(auth) },
      body: {
        content: messageValue,
        order_id: orderId,
      },
    });

    setMessages((prevMessages) => [...prevMessages, newMessage]); // Update local state
    setMessageValue("");
    scrollToBottom();
  } catch (error) {
    console.error("Failed to send message:", error);
  }
```